src/cplus_plugin/lib/constant_raster.py

```diff
@@ -0,0 +1,836 @@
+            )
+        else:
+            # Fallback to flat structure if metadata_id not provided
+            actual_output_dir = context.output_dir or os.path.join(
+                os.path.expanduser("~"), "cplus", "constant_rasters"
+            )
+
+        # Ensure output directory exists
+        if actual_output_dir and not os.path.exists(actual_output_dir):
+            os.makedirs(actual_output_dir, exist_ok=True)
+            if feedback:
+                feedback.pushInfo(f"Created output directory: {actual_output_dir}")
+
+        # Create timestamped session folder for this raster creation session
+        timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
+        session_dir = os.path.join(actual_output_dir, timestamp)
+        os.makedirs(session_dir, exist_ok=True)
+        if feedback:
+            feedback.pushInfo(f"Created session directory: {session_dir}")
+
+        created_rasters = []
+        processing_context = QgsProcessingContext()
+
+        # Process each component
+        total_components = len(enabled_components)
+        for idx, component in enumerate(enabled_components):
+            if feedback and feedback.isCanceled():
+                raise QgsProcessingException("Processing canceled by user")
+
+            # Update progress
+            if feedback:
+                progress = int((idx / total_components) * 100)
+                feedback.setProgress(progress)
+                component_name = (
+                    component.component.name
+                    if component.component and hasattr(component.component, "name")
+                    else component.component_id
+                )
+                feedback.pushInfo(
+                    f"Processing {idx + 1}/{total_components}: {component_name}"
+                )
+
+            # Get the constant value
+            if not component.value_info:
+                if feedback:
+                    feedback.pushWarning(
+                        f"Component {component.component_id} has no value_info, skipping"
+                    )
+                continue
+
+            absolute_value = component.value_info.absolute
+
+            if feedback:
+                feedback.pushInfo(f"Component value: {absolute_value}")
+                feedback.pushInfo(f"Input range: {input_min} - {input_max}")
+                feedback.pushInfo(
+                    f"Normalization range: {collection.min_value} - {collection.max_value}"
+                )
+
+            # TWO-STEP NORMALIZATION:
+
+            # Step 1: Normalize input value to [0,1] using input_range
+            if input_max != input_min:
+                normalized_0_1 = (absolute_value - input_min) / (input_max - input_min)
+                # Clamp to [0,1] range
+                normalized_0_1 = max(0.0, min(1.0, normalized_0_1))
+            else:
+                # Edge case: input range is zero, normalize to 0.5
+                normalized_0_1 = 0.5
```

Comment:
@osundwajeff in such a case, we should skip instead of providing a default value. See preceding comment on checking whether input_min == input_max